src/main/java/org/candlepin/async/tasks/RevokeEntitlementsJob.java

```diff
@@ -0,0 +1,169 @@
+import org.candlepin.async.JobConfigValidationException;
+import org.candlepin.async.JobConstraints;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.config.Configuration;
+import org.candlepin.controller.PoolService;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+import org.candlepin.model.activationkeys.ActivationKeyCurator;
+
+import com.google.common.collect.Iterables;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+/**
+ * An async job responsible for revoking all entitlements for all non-manifest consumers that belong to a
+ * specified owner. This job also removes activation key pools for the same owner.
+ */
+public class RevokeEntitlementsJob implements AsyncJob {
+    private static final Logger log = LoggerFactory.getLogger(RevokeEntitlementsJob.class);
+    private static final int MAX_RETRY_COUNT = 5;
+
+    public static final String JOB_KEY = "EntitlementRevokingJob";
+    public static final String JOB_NAME = "Entitlement Revoking Job";
+    public static final String CFG_BATCH_SIZE = "batch_size";
+    public static final String OWNER_KEY = "owner_key";
+
+    private Configuration config;
+    private ConsumerCurator consumerCurator;
+    private OwnerCurator ownerCurator;
+    private PoolService poolService;
+    private ActivationKeyCurator activationKeyCurator;
+
+    @Inject
+    public RevokeEntitlementsJob(Configuration config, ConsumerCurator consumerCurator,
+        OwnerCurator ownerCurator, PoolService poolService, ActivationKeyCurator activationKeyCurator) {
+        this.config = Objects.requireNonNull(config);
+        this.consumerCurator = Objects.requireNonNull(consumerCurator);
+        this.ownerCurator = Objects.requireNonNull(ownerCurator);
+        this.poolService = Objects.requireNonNull(poolService);
+        this.activationKeyCurator = Objects.requireNonNull(activationKeyCurator);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments jobArguments = context.getJobArguments();
+        String ownerKey = jobArguments.getAsString(OWNER_KEY);
+
+        log.info("{} has started for owner {}", JOB_KEY, ownerKey);
+        Owner owner = ownerCurator.getByKey(ownerKey);
+        if (owner == null) {
+            log.info("Unable to find owner for owner key \"{}\"", ownerKey);
+            context.setJobResult("Unable to find owner for owner key \"%s\"", ownerKey);
+            return;
+        }
+
+        int batchSize = getBatchSize();
+
+        int removedPools = activationKeyCurator.removeActivationKeyPools(ownerKey);
+        log.debug("Removed {} activation key pools", removedPools);
```

Comment:
I'm wondering if this really should be its own job so that each job has one concern and can possibly be used for other situations other than an owner changing from entitlement mode to SCA mode.